code/modules/mining/miningCharge.dm

```diff
@@ -0,0 +1,118 @@
+//MINING CHARGE: Slap it in rocks to cause a controlled explosion. Can be emagged to slap on other things.
+/obj/item/device/miningCharge
+	name = "standard mining charge"
+	desc = "A pyrotechnical device used to cause controlled explosions for digging tunnels without manual labor. It can only be attached to rocks and mineral deposits."
+	w_class = 2
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "miningCharge"
+	item_state = "electronic"
+	throw_speed = 3
+	throw_range = 5
+	slot_flags = SLOT_BELT
+	var/detonating = 0 //If the charge is currently primed
+	var/safety = 1 //If the charge can be put on things other than rocks
+	var/explosionPower = 2 //The power of the explosion; larger powers = bigger boom
+	var/atom/movable/putOn = null //The atom the charge is on
+	var/primedOverlay = null
+
+/obj/item/device/miningCharge/emag_act(mob/user)
+	if(!safety)
+		return
+	user << "<span class='warning'>You press the emag onto [src], disabling its safeties.</span>"
```

Comment:
use cryptographic sequencer for situations like these, we use emag as shorthand, but game text is more "professional" in a way and uses the full name.